src/components/EmailPreview.js

```diff
@@ -30,7 +31,9 @@ class EmailPreview extends React.Component {
         
             return (
                 
-                <tr className={`email ${this.state.read ? "read" : "unread"}`} onClick={markAsRead}> 
+                <tr className={`email ${this.state.read ? "read" : "unread"}`} onClick={() => {
+                                                                                                markAsRead();
+                                                                                                childToParent(this.state.id, this.state.starred, this.stateread);}}> 
```

Comment:
can you explain whats going on here?

Reply:
Im using a ternary operator for the className just for formatting purposes because read emails have normal font weight while unread emails have a bold font. then when the user clicks the email preview it calls two functions. markAsRead changes the state of the component, then childToParent (i will definitely change the name of it) is supposed to reflect those changes in the array of emails